homeassistant/components/homewizard/__init__.py

```diff
@@ -100,3 +99,43 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         await config_data.api.close()
 
     return unload_ok
+
+
+class HomeWizardEnergyData(TypedDict):
+    """Class for defining data in dict."""
+
+    device: Device
+    data: Data
+    state: State | None
+
+
+class HWEnergyDeviceUpdateCoordinator(DataUpdateCoordinator[HomeWizardEnergyData]):
+    """Gather data for the energy device."""
+
+    api: HomeWizardEnergy
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        host: str,
+    ) -> None:
+        """Initialize Update Coordinator."""
+
+        super().__init__(
+            hass, _LOGGER, name=DOMAIN, update_interval=timedelta(seconds=5)
```

Comment:
Could you make the update interval configurable or decrease it to 1 second so that it matches the frequency at which the HomeWizard meters provide the data?

Reply:
Would love this, but core requires a poll interval of at least 5 seconds.
If a user wants to configure poll interval, they should create an automation that calls `homeassistant.entity_update`

Reply:
And what about making it configurable where 5s is the default, but the user can set it to 1s? E.g. the current version of the P1 meter has this: https://github.com/home-assistant/core/blob/dev/homeassistant/components/dsmr/sensor.py#L67

Reply:
I had to remove polling configuration because the user can to disable polling in the settings and make an automation that calls `homeassistant.entity_update` when desired.

The DSMR integration is `local_push`. So it does not need to (and cannot) poll. It has a configurable option because it can flood the database. Maybe with the same arguments as https://github.com/home-assistant/core/pull/43057 we can make it configurable, but lets discuss that in an separate issue which you may open if that is what you want.